eng/common/TestResources/New-TestResources.ps1

```diff
@@ -127,6 +127,42 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5)
     }
 }
 
+# NewServicePrincipalWrapper creates an object from an AAD graph or Microsoft Graph service principal object type.
+# This is necessary to work around breaking changes introduced in Az version 7.0.0:
+# https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/
+function NewServicePrincipalWrapper([string]$subscription, [string]$resourceGroup, [string]$displayName)
+{
+    $servicePrincipal = Retry {
+        New-AzADServicePrincipal -Role "Owner" -Scope "/subscriptions/$SubscriptionId/resourceGroups/$ResourceGroupName" -DisplayName $displayName
+    }
+    $spPassword = ""
+    $appId = ""
+    if (Get-Member -Name "Secret" -InputObject $servicePrincipal -MemberType property) {
```

Comment:
Easier - and more robust, if the secret was unknown / empty, for example - to just do:

```powershell
if ($servicePrincipal.Secret) ...
```

PowerShell should never throw for missing properties or properties that themselves throw (by design). Even if you kept this, should pass `-ea SilentlyContinue` or this will generate error spew if missing (though, that behavior seems to have changed in pwsh w🤷‍♂️).